diesel/src/sql_types/mod.rs

```diff
@@ -341,6 +341,28 @@ pub struct Time;
 #[mysql_type = "Timestamp"]
 pub struct Timestamp;
 
+/// The JSON SQL type.  This type can only be used with `feature =
+/// "serde_json"`
+///
+/// For postgresql you should normally prefer [`Jsonb`](struct.Jsonb.html) instead,
+/// for the reasons discussed there.
+///
+/// ### [`ToSql`] impls
+///
+/// - [`serde_json::Value`]
+///
+/// ### [`FromSql`] impls
+///
+/// - [`serde_json::Value`]
+///
+/// [`ToSql`]: /serialize/trait.ToSql.html
+/// [`FromSql`]: /deserialize/trait.FromSql.html
+/// [`serde_json::Value`]: /../serde_json/value/enum.Value.html
+#[derive(Debug, Clone, Copy, Default, QueryId, SqlType)]
+#[postgres(oid = "114", array_oid = "199")]
+#[mysql_type = "String"]
+pub struct Json;
```

Comment:
I don't see a feature gate on this. I might be wrong and you probably want this struct to exist even without the feature (but not work).

Reply:
Sqltypes are intentionally not feature gated as they are independent from the actual `FromSql`/`ToSql` implementation. Not everyone needs to actually load/store such fields. Sometimes it's ok to just write queries that touches somehow those fields without loading/storing data there. Additionally even without the feature flag enabled those fields could be used by third parties to create their own `FromSql`/`ToSql` implementations for this (and also other types in general).